src/components/App.tsx

```diff
@@ -1917,8 +1917,8 @@ class App extends React.Component<AppProps, AppState> {
           this.state,
         );
         return [
-          viewportX - this.state.offsetLeft,
-          viewportY - this.state.offsetTop,
+          viewportX - this.state.offsetLeft * (2 - this.state.zoom.value),
+          viewportY - this.state.offsetTop * (2 - this.state.zoom.value),
```

Comment:
what's this `2 -` magic number?

Reply:
The 2 comes from reorganizing the formulas. When I changed (I believe corrected) `sceneCoordsToViewportCoords` to follow the same logic as `viewportCoordsToSceneCoords`, that had the knock-on effect on `getViewportCoords`. To be honest I don’t even understand why `getViewportCoords` needs the correction `-this.state.offset…` in the first place. That transformation already seems odd to me.

Reply:
> what's this `2 -` magic number?

I'll work on this PR and take a stab at solving #4421, #4418, #4417, and #4413

Similar to getViewportCoords in App.tsx I am thinking to create a `getSceneCoords` function on App.tsx and move the magic with numbers there. What do you think?

https://github.com/excalidraw/excalidraw/blob/8b5657e1cea96e78f9f9dc6fbd608a33a11310af/src/components/App.tsx#L1911-L1918